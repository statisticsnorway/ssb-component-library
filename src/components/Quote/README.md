Quote
========

> A component for highlighting quoted text.
### Usage


#### HTML

```html
<span class="ssb-quote">Default text here</span>

<span class="ssb-quote negative">Negative text here</span>
```

#### React

```jsx harmony
<Quote>Default text here</Quote>

<Quote negative>Negative text here</Quote>


```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| negative | Optional boolean | Styling for dark theme |
