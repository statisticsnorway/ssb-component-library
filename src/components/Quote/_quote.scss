.ssb-quote {
  @include open-sans;
  background-image: linear-gradient(120deg, $ssb-green-1 0%, $ssb-green-1 100%);
  background-position: 0 100%;
  background-repeat: no-repeat;
  background-size: 100% 50%;
  font-size: 20px;
  font-style: normal;
  letter-spacing: normal;
  line-height: 1.7;

  &.negative {
    background-image: linear-gradient(120deg, $ssb-green-5 0%, $ssb-green-5 100%);
    color: $ssb-white;
  }
}
