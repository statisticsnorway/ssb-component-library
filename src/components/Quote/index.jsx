import React from 'react';
import PropTypes from 'prop-types';

const Quote = ({ children, negative }) => (
	<span className={`ssb-quote${negative ? ' negative' : ''}`}>{ children }</span>
);

Text.defaultProps = {
	negative: false,
	small: false,
};

Text.propTypes = {
	children: PropTypes.node.isRequired,
	negative: PropTypes.bool,
};

export default Quote;
