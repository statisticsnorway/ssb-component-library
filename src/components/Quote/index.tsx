import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface QuoteProps {
  children: ReactNode
  className?: string
  negative?: boolean
}

const Quote: React.FC<QuoteProps> = ({ children, className, negative = false }) => (
  <span className={classNames('ssb-quote', negative && 'negative', className)}>{children}</span>
)

export default Quote
