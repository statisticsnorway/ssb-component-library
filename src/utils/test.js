import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
