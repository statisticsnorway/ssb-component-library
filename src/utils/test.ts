import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
