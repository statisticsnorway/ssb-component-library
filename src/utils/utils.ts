// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
