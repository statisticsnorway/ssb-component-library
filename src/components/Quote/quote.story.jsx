import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Quote from './index';

const negativeStyle = {
	backgroundColor: '#274247',
	padding: '100px',
};

storiesOf('Quote', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{padding: '100px'}}>
			<Quote>
				«Ser du etter en hytte med slik beliggenhet, må du regne med å måtte ut med en bra slump med penger hvis du ikke er så heldig å arve – selv om også dét kan være dyrt.»
			</Quote>
		</div>
	))
	.add('Negative', () => (
		<div style={negativeStyle}>
			<Quote	negative>
				«Ser du etter en hytte med slik beliggenhet, må du regne med å måtte ut med en bra slump med penger hvis du ikke er så heldig å arve – selv om også dét kan være dyrt.»
			</Quote>
		</div>
	));
