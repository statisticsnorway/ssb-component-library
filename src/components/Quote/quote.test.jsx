import React from 'react'
import { render } from '../../utils/test'
import Quote from './index'

describe('Quote component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Quote>«Quote»</Quote>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Toggles classNames correctly', () => {
    const { asFragment } = render(<Quote negative>«Quote»</Quote>)
    expect(asFragment()).toMatchSnapshot()
  })
})
