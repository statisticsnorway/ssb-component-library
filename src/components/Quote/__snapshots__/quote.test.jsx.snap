// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component Matches the snapshot 1`] = `
<DocumentFragment>
  <span
    class="ssb-quote"
  >
    «Quote»
  </span>
</DocumentFragment>
`;

exports[`Quote component Toggles classNames correctly 1`] = `
<DocumentFragment>
  <span
    class="ssb-quote negative"
  >
    «Quote»
  </span>
</DocumentFragment>
`;
